### PR TITLE
[Hotfix] [ENG-2917]Add new metadata types for Character Lab

### PR DESCRIPTION
**Pull request by futa-ikeda**
- Ticket: [ENG-2917]
- Feature flag: n/a

## Purpose
- Add new fields for Character Lab ( Study Design and School Type) on Collections Discover page and Collection Submission page
  - Users should be able to filter by these new fields and see them visible on Collection search cards on the discover page

## Summary of Changes
- Add new fields and their choices to collection and collected metadata model
- Added new filter/facet components for the new fields
- Updates to mirage to support new fields filtering

## Screenshots
- Discover page (osf.io/collections/<collection_id>/discover)
Screen Shot 2021-06-17 at 1 36 39 PM

- Submission page  (osf.io/collections/<collection_id>/submit)

Screen Shot 2021-06-17 at 1 38 53 PM

## Side Effects
- NA

## QA Notes
- This PR adds Study Design and School Type to Collections in preparation for Character Lab's new collection going live. Users should be able to add these new fields when they submit a project to character lab's collection and also be able to filter by these new types on the discover page.

[ENG-2917]: https://openscience.atlassian.net/browse/ENG-2917

### CODE REVIEW

**Comment by brianjgeiger** [comment]
@futa-ikeda What about submitting a new collection?

**Comment by futa-ikeda** [comment]
> @futa-ikeda What about submitting a new collected item?

The two new fields will be in the submission page (/collections/studyswap/submit) as long as the collection model has studyDesignChoices and schoolTypeChoices. I'll add some screenshots to help with the PR description

**Comment by brianjgeiger** [comment]
Awesome, thanks!

**Review by brianjgeiger — APPROVED** [review]

**Comment by brianjgeiger** [comment]
This is available on develop, so could be tested on staging2 at https://staging2.osf.io/collections/characterlab/discover if someone wanted to. Otherwise I'll verify there after it's merged.

File: lib/collections/addon/components/discover-page/active-filters/study-design/template.hbs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{{#each @context.activeFilter as |item|}}
+    <DiscoverPage::ActiveFilters::Filter
+        @item={{item}}
+        @text={{item}}
+        @ariaLabel={{t 'collections.discover_page.active_filters.remove_study_design'}}
+        @removeFilterItem={{action this.removeFilterItem item}}
```
Comment (fabmiz) [inline lib/collections/addon/components/discover-page/active-filters/study-design/template.hbs:6]:
```suggestion
        @removeFilterItem={{fn this.removeFilterItem item}}
```

File: lib/collections/addon/components/discover-page/facets/school-type/component.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import Collection from 'ember-osf-web/models/collection';
+import SearchFacetChecklist from '../checklist/component';
+
+class SchoolType extends SearchFacetChecklist {
+    get modelAttribute(): keyof Collection {
```
Comment (fabmiz) [inline lib/collections/addon/components/discover-page/facets/school-type/component.ts:5]: TS can infer the correct return type so not needed.
```suggestion
    get modelAttribute() {
```

File: lib/collections/addon/components/discover-page/facets/study-design/component.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import Collection from 'ember-osf-web/models/collection';
+import SearchFacetChecklist from '../checklist/component';
+
+class StudyDesign extends SearchFacetChecklist {
+    get modelAttribute(): keyof Collection {
```
Comment (fabmiz) [inline lib/collections/addon/components/discover-page/facets/study-design/component.ts:5]: ditto. correct type is inferable here.

**Review by fabmiz — COMMENTED** [review]
Looks great 💯 minor suggestions

**Pull request merged**